kotlinx-coroutines-core/common/src/internal/LimitedDispatcher.kt

```diff
@@ -45,61 +44,37 @@ internal class LimitedDispatcher(
         return super.limitedParallelism(parallelism)
     }
 
-    override fun run() {
-        var fairnessCounter = 0
-        while (true) {
-            val task = queue.removeFirstOrNull()
-            if (task != null) {
-                try {
-                    task.run()
-                } catch (e: Throwable) {
-                    handleCoroutineException(EmptyCoroutineContext, e)
-                }
-                // 16 is our out-of-thin-air constant to emulate fairness. Used in JS dispatchers as well
-                if (++fairnessCounter >= 16 && dispatcher.isDispatchNeeded(this)) {
-                    // Do "yield" to let other views to execute their runnable as well
-                    // Note that we do not decrement 'runningWorkers' as we still committed to do our part of work
-                    dispatcher.dispatch(this, this)
-                    return
-                }
-                continue
-            }
-
-            synchronized(workerAllocationLock) {
-                runningWorkers.decrementAndGet()
-                if (queue.size == 0) return
-                runningWorkers.incrementAndGet()
-                fairnessCounter = 0
```

Comment:
The behavior of this line of code was lost in the process. It can be restored but this wouldn't look as nice. Is it right that the gains here are marginal?